test/ParallelIOTest.cpp

```diff
@@ -523,4 +524,106 @@ TEST_CASE( "close_iteration_test", "[parallel]" )
+    // filling some values: 0, 1, ...
+    std::iota( E_x_data.begin(), E_x_data.end(), 0.);
+
+    // open a parallel series
+    Series series( name, Access::CREATE, MPI_COMM_WORLD );
+
+    // in HiPACE, ranks write one-by-one to a "swiped" step, overlapping
+    // each other in time;
+    int const last_step = 100;
+    int const my_first_step = i_mpi_rank * int(local_Nz);
+    int const all_last_step = last_step + (i_mpi_size - 1) * int(local_Nz);
+    for( int first_rank_step = 0; first_rank_step < all_last_step; ++first_rank_step )
+    {
+        // first_rank_step: this step will "lead" the opening of an output step
+        // step on the local rank
+        int const step = first_rank_step - my_first_step;
+
+        if( verbose )
+            std::cout << "[" << i_mpi_rank << "] " <<
+                "step: " << step << " | first_ranks_step: " << first_rank_step << std::endl;
+        // do we start writing to a new step?
+        bool const start_new_output_step =
+            std::find(iterations.begin(), iterations.end(), first_rank_step) != iterations.end();
+        // are we just about to finish writing to a step?
+        // TODO; if we detect this, we can collectively call `it.close()` after storeChunk/flush()
+
+        // collectively: create a new iteration and declare records we want to write
+        if( verbose )
+            std::cout << "[" << i_mpi_rank << "] " <<
+                "start_new_output_step: " << start_new_output_step << std::endl;
+        if( start_new_output_step && false ) // looks like we don't even need that :)
+        {
+            auto it = series.iterations[first_rank_step];
+            auto E = it.meshes["E"]; // record
+            auto E_x = E["x"]; // record component
+            auto dataset = io::Dataset(
+                io::determineDatatype< precision >( ),
+                {global_Nz, global_Nx});
+            E_x.resetDataset(dataset);
+        }
+
+        // has this ranks started computations yet?
+        if( step < 0 ) continue;
+        // has this ranks stopped computations?
+        if( step > last_step ) continue;
+        // does this rank contribute to with output currently?
+        bool const rank_in_output_step =
+            std::find(iterations.begin(), iterations.end(), step) != iterations.end();
+        if( !rank_in_output_step ) continue;
+
+        // now we write (parallel, independent I/O)
+        auto it = series.iterations[step];
+        auto E = it.meshes["E"]; // record
+        auto E_x = E["x"];       // record component
+
+        auto dataset = io::Dataset(
+            io::determineDatatype< precision >( ),
+            {global_Nz, global_Nx});
+        E_x.resetDataset(dataset);
+
+        Offset chunk_offset = {local_Nz * mpi_rank, 0};
+        Extent chunk_extent = {local_Nz, global_Nx};
+        E_x.storeChunk(
+            io::shareRaw(E_x_data),
+            chunk_offset, chunk_extent);
+        series.flush();
```

Comment:
flush should be MPI collective and hangs in the app at this point

Reply:
Ah jeeze, I forgot that of course there is other, app-related collective/blocking I/O in this loop...